File: frontend/src/types/operations.ts
export type OperationAlertCategory = "INVENTORY" | "ORDER" | "SUPPORT" | "REFUND";
export type OperationAlertSeverity = "CRITICAL" | "WARNING";

export interface OperationAlert {
  id: string;
  category: OperationAlertCategory;
  severity: OperationAlertSeverity;
  title: string;
  description: string;
  source_at: string;
  href: string;
}

export interface OperationAlertPage {
  items: OperationAlert[];
  total: number;
  limit: number;
  offset: number;
  counts: {
    total: number;
    critical: number;
    warning: number;
  };
}

export interface OperationAlertFilters {
  category?: OperationAlertCategory | null;
  severity?: OperationAlertSeverity | null;
  search?: string;
  limit?: number;
  offset?: number;
}

File: backend/app/schemas/operations.py
from datetime import datetime
from typing import Literal

from pydantic import BaseModel


AlertCategory = Literal["INVENTORY", "ORDER", "SUPPORT", "REFUND"]
AlertSeverity = Literal["CRITICAL", "WARNING"]


class OperationAlert(BaseModel):
    id: str
    category: AlertCategory
    severity: AlertSeverity
    title: str
    description: str
    source_at: datetime
    href: str


class OperationAlertCounts(BaseModel):
    total: int
    critical: int
    warning: int


class OperationAlertPage(BaseModel):
    items: list[OperationAlert]
    total: int
    limit: int
    offset: int
    counts: OperationAlertCounts
